src/entities/nestedJoin/types.ts

```diff
@@ -0,0 +1,48 @@
+import { Address, Hex } from '../../types';
+import { Slippage } from '../slippage';
+import { Token } from '../token';
+import { TokenAmount } from '../tokenAmount';
+
+export type NestedJoinInput = {
+    amountsIn: {
```

Comment:
The main issue with `TokenAmount` is that it has attributes that could cause some confusion, such as `wrapped`.
The minimal info we need from the user is token `address` and `rawAmount`, because we can get `decimals` from the token provided by the api (i.e. poolState), but if I remember correctly @danielmkm wanted to keep decimals there to keep consistency.
In the end that was the format that made most sense.
Let me know if you think there's a better option.